std/stdio.d

```diff
@@ -2942,8 +2941,8 @@ $(D Range) that locks the file and allows fast writing to it.
                     else
                     {
                         char[4] buf = void;
-                        immutable len = encode(buf, c);
-                        foreach (i ; 0 .. len)
+                        immutable len = encode!(UseReplacementDchar.yes)(buf, c);
```

Comment:
Here, I changed this for two reasons,

1. to be in-line with the lines above which don't throw
2. DMD can forgo adding the exception-handling code if the function is marked as nothrow, which encode is when using replacement dchars

Reply:
We should never be silently accepting/converting garbage! `UseReplacementDchar.yes` needs to be used with a lot of caution. If this does what I think it does (replace any kind of bad UTF with a single replacement character), then it's a bad idea. See https://issues.dlang.org/show_bug.cgi?id=14519

Reply:
Well, I see you're just copying the above, which you've added in #5083 to replace `toUtf8` which did the same thing, but I think we need to change this. Silently ignoring errors and throwing away information is never a good idea.